src/util/message.h

```diff
@@ -44,10 +45,57 @@ struct messaget
     template <typename... Args>
     static void println(FILE *f, VerbosityLevel lvl, Args &&...args)
     {
-      if(lvl == VerbosityLevel::Error)
-        fmt::print(f, "ERROR: ");
-      fmt::print(f, std::forward<Args>(args)...);
-      fmt::print(f, "\n");
+    	     
+      	/** 
+		* Newly added code to address issue #886.
+			* It prepends a well formatted timestamp, file path, function name, line number and log level to each LOG message. 
+			* It is transparent to the existing logging function, i.e., log_message.
+			* TODO: It can be implemented using a separate function and invoked here and/or elsewhere when required. 		
+		*/		
+		//========Start===============================================================================================
+		struct tm _stime; //A time structure consisting of 9 members, such as tm_sec, tm_min, tm_hour, tm_mday, tm_mon, tm_year, etc. 
+		char _buffer[20]; //A variable to hold values of certain _stime members.
+		char _vrb[20]; //A variable to hold the log level. 
+		
+		time_t _currenttime = time(0);
+		localtime_r (&_currenttime, &_stime); //"localtime_r" works for POSIX-based systems. Use "localtime_s", otherwise. 
+		strftime (_buffer, sizeof(_buffer), "%Y-%m-%d %H:%M:%S", &_stime); //"strftime" is to format the date/time structure. 
+	   
+		switch(lvl){//To prepend the respective log level to each log message.  
+			case VerbosityLevel::Error:
+				strncpy(_vrb, "[ERROR]:", sizeof(_vrb));
+				break;
+			case VerbosityLevel::Result:
+				strncpy(_vrb, "[RESULT]:", sizeof(_vrb));
+				break;
+			case VerbosityLevel::Warning:
+				strncpy(_vrb, "[WARNING]:", sizeof(_vrb));
+				break;
+			case VerbosityLevel::Progress:
+				strncpy(_vrb, "[PROGRESS]:", sizeof(_vrb));
+				break;
+			case VerbosityLevel::Status:
+				strncpy(_vrb, "[STATUS]:", sizeof(_vrb));
+				break;
+			case VerbosityLevel::Debug:
+				strncpy(_vrb, "[DEBUG]:", sizeof(_vrb));
+				break;
+			default:
+				printf("Incorrect or no verbosity!\n");
```

Comment:
Anything else can be defaulted to Debug.

Reply:
Corrected, thanks